src/libostree/ostree-fetcher-soup.c

```diff
@@ -990,14 +941,24 @@ on_stream_read (GObject        *object,
         }
       else
         {
-          g_task_return_pointer (task,
-                                 g_strdup (pending->out_tmpfile),
-                                 (GDestroyNotify) g_free);
+          g_autofree char *uristring
+            = soup_uri_to_string (soup_request_get_uri (pending->request), FALSE);
+          g_autofree char *tmpfile_path =
+            ostree_fetcher_generate_url_tmpname (uristring);
+          if (!glnx_link_tmpfile_at (&pending->tmpf, GLNX_LINK_TMPFILE_REPLACE,
```

Comment:
Looks like we need to `chmod` to `0666` here before linking it in place.

```
+ mode=600
+ test 600 = 600
+ assert_not_reached 'Content object unreadable by others: 600'
+ fatal 'Content object unreadable by others: 600'
+ echo Content object unreadable by others: 600
Content object unreadable by others: 600
```

Reply:
Yep, thanks for digging through the logs before me 😄  Fixup ⬇️